gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
I actually think I slightly prefer what I just described for exactly the reasons you say. While it should be clear from the code that `self.num_workers` _must_ change or the loops would not terminate, it's perhaps clearest to just set a flag when we're attempting to change the number.

It's conceivable that `self.num_workers` could change between the loops so that it is first increased and then decreased, so if the edge case of logging the same number twice bothers us then your way is indeed better.